starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+
+	// 10% chance each to move north/east/south/west (This is extra smart)
+	if(decision <= 10){
+		move_north(ship_id);
+	}else if(decision <= 20){
+		move_east(ship_id);
+	}else if(decision <= 30){
+		move_south(ship_id);
+	}else if(decision <= 40){
+		move_west(ship_id);
+	}else{
+		// And a 60% chance to stay still (This is also extra smart)
+		// We could alternatively just not give the ship a move this turn
+		move_still(ship_id);
+	}
+}
+
+
+// There's something that random_move is forgetting... Let's avoid collisions!
+void naive_navigate(unsigned short ship_id){
+	unsigned char decision = rand() % 100, x = ships[ship_id].x, y = ships[ship_id].y;
+
+	// North is y - 1
+	// South is y + 1
+	// West is x - 1
+	// East is x + 1
+
+	// 10% chance each to move north/east/south/west (This is even smarter than before!)
+	if(decision <= 10 && tile_occupied(x, norm_y(y - 1)) == 0){
+		move_north(ship_id);
+	}else if(decision <= 20 && tile_occupied(norm_x(x + 1), y) == 0){
+		move_east(ship_id);
+	}else if(decision <= 30 && tile_occupied(x, norm_y(y + 1)) == 0){
+		move_south(ship_id);
+	}else if(decision <= 40 && tile_occupied(norm_x(x - 1), y) == 0){
+		move_west(ship_id);
+	}else{
+		// And a 60% chance to stay still (This is also smarter than before!)
+		move_still(ship_id);
+	}
+}
+// This file can safely be removed or replaced with your own code
+
+
+
+int main(int argc, char* argv[]){
+	char *c, inspiration_enabled_str[6];
+	unsigned char i, id, j, sca, scb, scc;
+	unsigned short s, ssa, ssb;
+	unsigned int sia;
+
+	// Firstly, we open a log file. If no log file is specified, it defaults to "MyBot.log"
+	if(argc != 2){
+		c = "MyBot.log";
+	}else{
+		c = argv[1];
+	}
+
+	log_file = fopen(c, "w");
+
+	fprintf(log_file, "Reading JSON\n");
+	fflush(log_file);
+
+	// The first input we get from the engine is some JSON-encoded constants. I'm just going to scanf this becuase the formatting is consistent
+	if(scanf("{\"CAPTURE_ENABLED\":%*5[^,],\"CAPTURE_RADIUS\":%*u,\"DEFAULT_MAP_HEIGHT\":%*u,\"DEFAULT_MAP_WIDTH\":%*u,\"DROPOFF_COST\":%hu,\"DROPOFF_PENALTY_RATIO\":%*u,\"EXTRACT_RATIO\":%hhu,\"FACTOR_EXP_1\":%*f,\"FACTOR_EXP_2\":%*f,\"INITIAL_ENERGY\":%*u,\"INSPIRATION_ENABLED\":%5[^,],\"INSPIRATION_RADIUS\":%hhu,\"INSPIRATION_SHIP_COUNT\":%hhu,\"INSPIRED_BONUS_MULTIPLIER\":%hhu.0,\"INSPIRED_EXTRACT_RATIO\":%hhu,\"INSPIRED_MOVE_COST_RATIO\":%hhu,\"MAX_CELL_PRODUCTION\":%*u,\"MAX_ENERGY\":%hu,\"MAX_PLAYERS\":%*u,\"MAX_TURNS\":%hu,\"MAX_TURN_THRESHOLD\":%*u,\"MIN_CELL_PRODUCTION\":%*u,\"MIN_TURNS\":%*u,\"MIN_TURN_THRESHOLD\":%*u,\"MOVE_COST_RATIO\":%hhu,\"NEW_ENTITY_ENERGY_COST\":%hu,\"PERSISTENCE\":%*f,\"SHIPS_ABOVE_FOR_CAPTURE\":%*u,\"STRICT_ERRORS\":%*5[^,],\"game_seed\":%*u}", &dropoff_cost, &extract_ratio, (char *)&inspiration_enabled_str, &inspiration_radius, &inspiration_ship_count, &inspired_bonus_multiplier, &inspired_extract_ratio, &inspired_move_cost_ratio, &max_energy, &max_turns, &move_cost_ratio, &ship_cost) < 12){
```

Comment:
This parses the game constants from the engine at the start of each game, I opted to do this rather than use a JSON parser. The comment above explains that this line does. This is core game engine parsing stuff, so I didn't think the method I used particularly mattered since anybody who wanted to deal with game engine input directly wouldn't use a starter kit.